Add __contains__ to DictMixin

pamelafox opened the pull request:
# Description

I discovered in application code that I could not call the "in" operator on an object that subclassed DictMixin, and that was because it is missing __contains__. It does have has_key, but that's considered deprecated in Python 3, so I don't want to use that in my app code.

I found a previous PR that added __contains__ to a single DictMixin, but there are quite a few copies of DictMixin in the monorepo, and most did not have it. So in addition the one I needed (in the blob client), I added it to all the other ones, and copied the same tests to all of them (taken from eventhub's test).

It seems like it'd be better if the repo didn't have so many copies of DictMixin, but I guess that'd require moving it to core, and perhaps there's a reason it's not there.

Not sure if I should update CHANGELOG for this?

# All SDK Contribution checklist:
- [X] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [X] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [X] Title of the pull request is clear and informative.
- [X] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [X] Pull request includes test coverage for the included changes.

azure-sdk commented:
**API change check**

APIView has identified API level changes in this PR and created following API reviews.

[azure-servicebus](https://apiview.dev/Assemblies/Review/8171360bc8f849caade5d6a32cfa6a77)

annatisch commented on sdk/servicebus/azure-servicebus/azure/servicebus/amqp/_amqp_message.py:
```diff
@@ -43,6 +43,9 @@ def __ne__(self, other: Any) -> bool:
     def __str__(self) -> str:
         return str({k: v for k, v in self.__dict__.items() if not k.startswith("_")})
 
+    def __contains__(self, key):
```
Looks like this mixin has been updated for type hints in Storage yet, but should be for ServiceBus.
```suggestion
    def __contains__(self, key: str) -> bool:
```

annatisch reviewed (commented):
Thank you so much for this!
I missed this in the original Storage implementation years ago (which I think then got reused in other SDKs, multiplying my mistake) - really appreciate your taking the time to fix this!

kristapratico commented:
@annatisch @kashifkhan @swathipil @l0lawrence @vincenttran-msft @jalauzon-msft 

Can we accept this PR? Any concerns?

swathipil commented on sdk/servicebus/azure-servicebus/tests/test_dictmixin_mgmt.py:
```diff
@@ -0,0 +1,76 @@
+from azure.servicebus.management._models import DictMixin
+
+
+class TestDictMixin:
```
This test file should be move to the `tests/mgmt_tests` folder.

swathipil reviewed (approved):
one small comment, but everything else lgtm! Thanks @pamelafox!

jalauzon-msft commented on sdk/storage/azure-storage-blob/tests/test_dictmixin.py:
```diff
@@ -0,0 +1,76 @@
+        key = "testkey"
+        value = "testval"
+        model.__setitem__(key, value)
+        #assert model.__contains__(key)
```
Was this meant to be uncommented?

jalauzon-msft commented on sdk/storage/azure-storage-file-datalake/tests/test_dictmixin.py:
```diff
@@ -0,0 +1,76 @@
+from azure.storage.filedatalake._models import DictMixin
+
+
+class TestDictMixin:
```
Since this is shared code (in our `_shared` folders) it probably does not need to be tested in every package. You could remove the test files for every package and leave only Blob just to reduce copy/pasted code.

jalauzon-msft reviewed (approved):
Two minor comments for Storage, otherwise looks good.

swathipil commented:
/azp run python - servicebus - tests

swathipil commented on sdk/servicebus/azure-servicebus/tests/test_dictmixin_amqp.py:
```diff
@@ -0,0 +1,76 @@
+        key = "testkey"
+        value = "testval"
+        model.__setitem__(key, value)
+        #assert model.__contains__(key)
```
missed this, sorry! same comment as @jalauzon-msft - should this be uncommented here and in the other sb test file?

pamelafox replied:
I copied the code verbatim from the eventhub test to keep the code in sync: (which included commented out line)
https://github.com/Azure/azure-sdk-for-python/blob/0e6b75a44ef4130aae3de32bfff759d332d98f27/sdk/eventhub/azure-eventhub/tests/unittest/test_dict_mixin.py#L17

I could either uncomment or remove it from all tests, however. Up to you if you think its enough to check haskey or if it should also check contains.

swathipil replied:
ah, thanks for the context! It seems I originally commented this out since the `key in model` check right below calls `__contains__` as well, so it doesn't need to be called explicitly. Sorry about the confusion!

Would you be able to remove line 18 from all test files you've added here and change line 19 to the following?
`assert key in model    # calls __contains__`

This should help us avoid confusion in the future. 🙂

pamelafox commented:
I believe I've made all the requested changes:

* replaced the commented out line of code everywhere with a comment next to "in" line instead
* removed most of the copies of the text for the _shared modules, leaving only the blob one
* moved the mgmt test into mgmt_tests
*

swathipil commented:
Thanks for making all the changes @pamelafox - will merge as soon as everything is green!
